cmd/migrate/db/mariadb.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+package db
+
+import (
+	"fmt"
+	"os"
+
+	"database/sql"
+
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/scribe-org/scribe-server/cmd/migrate/types"
+	"gopkg.in/yaml.v2"
+)
+
+// LoadConfig reads and unmarshals the YAML config file.
+func LoadConfig(path string) (*types.Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %v", err)
+	}
+	var config types.Config
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config: %v", err)
+	}
+	return &config, nil
+}
```

Comment:
`LoadConfig()` perhaps make sense somewhere else, like in a `cmd/migrate/config.go` file along with the `Config` struct that it uses :thinking: